airbyte-test-utils/src/main/java/io/airbyte/test/utils/AirbyteAcceptanceTestHelper.java

```diff
@@ -0,0 +1,721 @@
+import io.airbyte.api.client.model.generated.SourceDefinitionRead;
+import io.airbyte.api.client.model.generated.SourceDiscoverSchemaRequestBody;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.commons.util.MoreProperties;
+import io.airbyte.db.Database;
+import io.airbyte.test.airbyte_test_container.AirbyteTestContainer;
+import io.airbyte.workers.temporal.TemporalUtils;
+import io.airbyte.workers.temporal.scheduling.ConnectionManagerWorkflow;
+import io.airbyte.workers.temporal.scheduling.state.WorkflowState;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.temporal.client.WorkflowClient;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import java.io.File;
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.jooq.JSONB;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SQLDialect;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.utility.MountableFile;
+
+public class AirbyteAcceptanceTestHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AirbyteAcceptanceTestHelper.class);
+
+  private static final String DOCKER_COMPOSE_FILE_NAME = "docker-compose.yaml";
+  // assume env file is one directory level up from airbyte-tests.
+  private final static File ENV_FILE = Path.of(System.getProperty("user.dir")).getParent().resolve(".env").toFile();
+
+  private static final String SOURCE_E2E_TEST_CONNECTOR_VERSION = "0.1.1";
+  private static final String DESTINATION_E2E_TEST_CONNECTOR_VERSION = "0.1.1";
+
+  private static final Charset UTF8 = StandardCharsets.UTF_8;
+  private static final boolean IS_KUBE = System.getenv().containsKey("KUBE");
+  private static final boolean IS_MINIKUBE = System.getenv().containsKey("IS_MINIKUBE");
+  private static final boolean IS_GKE = System.getenv().containsKey("IS_GKE");
+  private static final boolean IS_MAC = System.getProperty("os.name").startsWith("Mac");
+  private static final boolean USE_EXTERNAL_DEPLOYMENT =
+      System.getenv("USE_EXTERNAL_DEPLOYMENT") != null && System.getenv("USE_EXTERNAL_DEPLOYMENT").equalsIgnoreCase("true");
```

Comment:
Interesting point. I was thinking the opposite, that these states shouldn't be directly affecting how the acceptance tests are written/used, so that it would be better to keep this state closer to where it's used, in this case the internals of interacting with the containers.

Both arguments make sense to me. I guess ultimately my assumption was that readers shouldn't care about these things when thinking through unit tests, but now that I type it out, that seems incorrect. Probably best to let users decide for themselves if they care about it or not.

Reply:
Sorry if I wasn't clear - I think/like keeping this private in the helper class. The main change I was getting at is moving the env checking logic into a private function called when a new helper class is created instead of part of static variables.

Reply:
oh okay, yeah that makes sense to me. I didn't think too hard when separating these out into the helper/harness class. Mainly just copy and pasting.